[Chia Ann Yao Eugene] iP

eugenechiaay opened the pull request:
### Gene

> "I've eaten nine birthday cakes, and I still feel empty." - Gene Belcher ([source](https://www.bustle.com/articles/99867-9-quotes-that-prove-gene-from-bobs-burgers-is-the-most-underrated-belcher-of-them-all))

Gene is a Duke clone which helps you to keep track of your tasks. Some features of Gene include:

- Adding tasks
- Marking/Unmarking 
- ~~Deleting~~ tasks

All you need to do is,

1. Download it from [here](https://github.com/eugenechiaay/ip/releases). 
2. Double-click it.
3. Add your tasks :heavy_exclamation_mark::heavy_exclamation_mark:

It's also **FREE**

Coming _soon_:

- [ ] Managing deadlines
- [ ] Reminders
- [X] Finding tasks

If you are a Java programmer, you can use it to practice java too. Here's the `main` method:

```
public static void main(String[] args) {
        (new Gene("gene.txt")).run();
    }
```

Commit:
Add Gradle support

eugenechiaannyao committed:
Added Identity class. Identity contains logo, greeting as well as goodbyes. Identity also includes static exitMatch and Goodbye methods

eugenechiaannyao committed:
Add TaskList class and updated functionality to add add action

eugenechiaannyao committed:
Added Task, parent of markedtask and unmarkedtask

eugenechiaannyao committed:
Added mark and umark functionality for tasks. Duke now listens to unmark and mark commands

eugenechiaannyao committed:
Added functionality for creating new types of event, deadline and todo tasks

eugenechiaannyao committed:
fixed unmarking and marking bugs

eugenechiaannyao committed:
auto ui testing!!

eugenechiaannyao committed:
exceptions done

eugenechiaannyao committed:
Added delete task function

eugenechiaannyao committed:
Updated duke main to gene, new logo, new structure -> removed makr and unmark and use Evet, Deadline and todo task. update exception mechanism

eugenechiaannyao committed:
Added create directory and file mechanism when gene starts up

eugenechiaannyao committed:
Added file read and file write fuctionality but just for add task commands

eugenechiaannyao committed:
added updates to file actions for mark and unmark comands

eugenechiaannyao committed:
fixed bugs for additon of tasks with multi word descriptions and deadlines

eugenechiaannyao committed:
Temp handling of deleteing empty file errors

eugenechiaannyao committed:
Added datetime recognition and printing in very user friendly and informational format

eugenechiaannyao committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/TaskList.java

ongkimlai commented on src/main/java/DeadlineTask.java:
```diff
@@ -0,0 +1,47 @@
+    private boolean markedStatus;
+    private LocalDateTime deadline;
+
+    public DeadlineTask (String taskTitle, LocalDateTime deadline) {
```
Not a naming violation but I noticed an extra space before (String taskTitle, ...) here 😄.
```suggestion
    public DeadlineTask(String taskTitle, LocalDateTime deadline) {
```

ongkimlai reviewed (commented):
Overall good naming conventions, followed the guideline well. 👍

eugenechiaannyao committed:
Midway restructuring, added commands and parser. Now moving on to storage and ui and remodelling of tasklist

eugenechiaannyao committed:
Resolved kinks when changing structure

eugenechiaannyao committed:
Further changes while rafactoring

eugenechiaannyao committed:
Fix bugs

eugenechiaannyao committed:
fixed most bugs after refactoring

eugenechiaannyao committed:
Moved files into packages. All files are in gene package. Within gene, command, component, exceptation and task pakages are craeted with relevant files organised accordingly

eugenechiaannyao committed:
not sure what im doing tbh

eugenechiaannyao committed:
Merge branch 'add-gradle-support' of https://github.com/eugenechiaay/ip

# Conflicts:
#	text-ui-test/runtest.sh

eugenechiaannyao committed:
ttrying to get gradle to work

eugenechiaannyao committed:
Finally got unit testing to work. Added tests for addCommand and eventTask

eugenechiaannyao committed:
added javadocs to all commands and components

eugenechiaannyao committed:
Fixed java coding conventions

eugenechiaannyao committed:
java docs

cwnm commented on src/main/java/gene/Gene.java:
```diff
@@ -0,0 +1,75 @@
+ * @since 2022-01-12
+ */
+public class Gene {
+    private final Ui geneUi;
```
I like the naming of variables here, very easy to follow.

cwnm commented on src/main/java/gene/Gene.java:
```diff
@@ -0,0 +1,75 @@
+     * @return void
+     */
+    private void run() {
+        geneUi.printGreeting();
```
Maybe you can consider removing the spaces for a more compact code?

cwnm commented on src/main/java/gene/command/AddCommand.java:
```diff
@@ -0,0 +1,231 @@
+            //Accept string
+            //Parse string to datetime, use formatter here to change reading style
+            //when printing, check date time and print format.
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/MM/yyyy HHmm");
```
Is there another way of inputting date eg. YYYY/MM/DD?

teominghuang commented on src/main/java/EventTask.java:
```diff
@@ -1,10 +1,14 @@
 public class EventTask extends Task {
     private final String taskTitle;
     private final String taskType = "E";
     private boolean markedStatus;
```
Great adherence to the coding standards thus far, just something small that I noticed, take note that boolean variables should be revised to sound boolean. :)

teominghuang commented on src/main/java/TaskList.java:
```diff
@@ -58,17 +62,24 @@ public void addEvent(String taskKey) {
+            System.out.println(deadline);
+            Task newTask = new EventTask(taskTitle, LocalDateTime.parse(deadline, formatter));
+            this.itemList.add(newTask);
+            System.out.println(
```
For this portion, what will happen if, for any reason, the addition of the task fails and an error is not thrown? I am just afraid that the system will continue printing out the success statement despite failing to add the task. May I suggest that a possible structure would be to return a boolean value for a success addition, and nest the success message within the if-statement of the addTask.

teominghuang commented on src/main/java/TaskList.java:
```diff
@@ -97,17 +108,26 @@ public void addDeadline(String taskKey) {
+            Task newTask = new DeadlineTask(taskTitle, LocalDateTime.parse(deadline, formatter));
+            this.itemList.add(newTask);
+            System.out.println(
+                    "----------------------------" +
```
I like how the variable names have been clear thus far! Just nit-picking, In this case, I think the hyphenated values can be further abstracted into a String constant to reduce repetition and reduce the length of the print statement. Otherwise, this is well-done!

teominghuang reviewed (changes requested):
I like the clear and structured aspect of your code-base. It is clear that a lot of thought was put into it. I think there are a few minute junctures where the logic of the method can be improved, and String and Integer constants can be utilised to elevate the project even more, what do you think? All in all, I think it is a job well-done, I am excited to see how this project will evolve over the semester.

eugenechiaannyao committed:
added find functions. Added find command, and find match in parser

eugenechiaannyao committed:
Merge branch 'branch-Level-9' into branch-A-CodingStandard

eugenechiaannyao committed:
Merge branch 'branch-Level-9'

eugenechiaannyao committed:
Merge branch 'branch-A-CodingStandard'

eugenechiaannyao committed:
A-CheckStyle

eugenechiaannyao committed:
Integrated GUI and gene logic albeit with file writing issues :(

eugenechiaannyao committed:
Still have file writing issues

eugenechiaannyao committed:
Fixed bugs with storage and file stuff. HAve NO IDEAhow...

eugenechiaannyao committed:
Added assertions

eugenechiaannyao committed:
Revert "Added assertions"

This reverts commit e052b520f449b5991b995b9f9147748dee3feb49.

eugenechiaannyao committed:
Assertions and fixed java style

eugenechiaannyao committed:
improved code quality

eugenechiaannyao committed:
no streams

Commit:
Merge pull request #2 from eugenechiaay/branch-A-CodeQuality

improved code quality

Commit:
Merge pull request #3 from eugenechiaay/branch-A-Streams

no streams

eugenechiaannyao committed:
Merge branch 'master' of https://github.com/eugenechiaay/ip

eugenechiaannyao committed:
Merge branch 'master' into branch-A-Assertions

Commit:
Merge pull request #4 from eugenechiaay/branch-A-Assertions

Branch a assertions

eugenechiaannyao committed:
yml CI

eugenechiaannyao committed:
Added support for Location

eugenechiaannyao committed:
Added Exception handling, better padding, better greetings. Added help command

eugenechiaannyao committed:
better GUi

eugenechiaannyao committed:
Help commad: Registers a help command if user needs help

Help command is case sensitive

Case sensitive but upon startup, the specific thing to key in for help is specified.

eugenechiaannyao committed:
Updated user guide and Ui.png

eugenechiaannyao committed:
upadted read me and fixed some bugs

Commit:
Set theme jekyll-theme-tactile

eugenechiaannyao committed:
Merge branch 'master' of https://github.com/eugenechiaay/ip

eugenechiaannyao committed:
deleted external library

eugenechiaannyao committed:
rename Ui.png.png to Ui.png
